elastic/kibana
# [serverless] Advanced settings - Search query bar

## Summary

This PR adds a Search query bar and a Category filter to the Settings application in `packages/kbn-management/settings/application` (which is currently integrated in serverless).

<details>
<summary>Screenshots</summary>

**Search query bar:**
<img width="1365" alt="Screenshot 2023-10-04 at 20 54 03" src="https://github.com/elastic/kibana/assets/59341489/c13dbc80-94e1-4163-89c1-7caba4f92c67">

<br>

**Parsing error:**
<img width="1239" alt="Screenshot 2023-10-05 at 10 16 54" src="https://github.com/elastic/kibana/assets/59341489/08cfaeb9-8715-484d-9c8f-7e26c16c3af8">

<br>

**Empty state:**
<img width="1365" alt="Screenshot 2023-10-04 at 20 54 31" src="https://github.com/elastic/kibana/assets/59341489/68a021b8-d10c-4892-abb5-382b35ef8b33">

<br>

**Category filter:**
<img width="1365" alt="Screenshot 2023-10-04 at 20 54 50" src="https://github.com/elastic/kibana/assets/59341489/7110dbdd-e2c6-44d0-9f5a-241c54492037">

</details>

### How to test:

1. Start Es with `yarn es serverless` and Kibana with `yarn serverless-{es/oblt/security}` (also possible to test in the Storybook from the CI build).
2. Go to Management -> Advanced settings
3. Verify that the Search query bar and the Category filter work as expected.
4. Verify that clicking the "clear search" link in each category panel clears the query.
5. Verify that when no settings match the query, the empty state is rendered and clicking the "clear search" link clears the query.

### Checklist
